### DISCUSSION TITLE
Some versions not updated (x.y.z-alpha) with sbt manager

### DISCUSSION THREAD

**Discussion by gaeljw** [renovatebot/renovate#22092] [Suggest an Idea]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Describe the bug

Manager: sbt (not sure if specific to sbt though)
Kind of dependency: Maven published library

I noticed that some dependencies are not getting Renovated even though Renovate do detect them.
I suspect this has something to do with their version number x.y.z-alpha.

For instance, the following is detected but do not get any updates (even though there any several versions above like 1.26.0-alpha, see https://mvnrepository.com/artifact/io.opentelemetry/opentelemetry-exporter-prometheus):

```
"io.opentelemetry" % "opentelemetry-exporter-prometheus" % "1.20.1-alpha"
```

While the folloing works fine (version 1.26.0 is suggested):

```
"io.opentelemetry" % "opentelemetry-sdk" % "1.25.0"
```

**Reproduction repository: https://github.com/gaeljw/renovate-sbt-otel-alpha**

### Relevant debug logs

```json
"sbt": [
      {
        "deps": [
          {
            "currentValue": "1.25.0",
            "currentVersion": "1.25.0",
            "datasource": "sbt-package",
            "depName": "io.opentelemetry:opentelemetry-sdk",
            "dependencyUrl": "https://repo.maven.apache.org/maven2/io/opentelemetry",
            "fixedVersion": "1.25.0",
            "isSingleVersion": true,
            "packageName": "io.opentelemetry:opentelemetry-sdk",
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "sourceUrl": "https://github.com/open-telemetry/opentelemetry-java",
            "versioning": "ivy",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "1.26.0",
                "newValue": "1.26.0",
                "newMajor": 1,
                "newMinor": 26,
                "updateType": "minor",
                "branchName": "renovate/io.opentelemetry-opentelemetry-sdk-1.x"
              }
            ]
          },
          {
            "currentValue": "1.20.1-alpha",
            "currentVersion": "1.20.1-alpha",
            "datasource": "sbt-package",
            "depName": "io.opentelemetry:opentelemetry-exporter-prometheus",
            "dependencyUrl": "https://repo.maven.apache.org/maven2/io/opentelemetry",
            "fixedVersion": "1.20.1-alpha",
            "packageName": "io.opentelemetry:opentelemetry-exporter-prometheus",
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "sourceUrl": "https://github.com/open-telemetry/opentelemetry-java",
            "versioning": "ivy",
            "warnings": [],
            "updates": []
          },
```

**Comment by gaeljw** [comment]
I actually have another project with same dependencies but using Maven and it works fine. So this look's like an issue with sbt specifically.

**Reply by rarkins** [reply]
Renovate won't update to unstable versions unless:
- it's the same major.minor.patch (e.g. updating from alpha.1 to alpha.2 of the same version)
- You set ignoreUnstable=false

**Reply by gaeljw** [reply]
Ok, I could go with `ignoreUnstable=false` for these projects but still the issue doesn't happen with Maven (maybe you'd like a reproduction repro showing it works with Maven?), thus there's different behavior here between sbt and Maven for the same dependencies.

**Reply by rarkins** [reply]
It would be useful to have a reproduction repo showing the same package/version behaving differently for maven and sbt.

**Reply by gaeljw** [reply]
Nevermind, indeed Maven behaves the same as sbt. Sorry for the false information.
